kotlin/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.kt

```diff
@@ -0,0 +1,354 @@
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geoanalysis.LocationViewshed
+import com.esri.arcgisruntime.geoanalysis.Viewshed
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.layers.ArcGISSceneLayer
+import com.esri.arcgisruntime.mapping.ArcGISScene
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Surface
+import com.esri.arcgisruntime.mapping.view.AnalysisOverlay
+import com.esri.arcgisruntime.mapping.view.Camera
+import com.esri.arcgisruntime.mapping.view.DefaultSceneViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.OrbitLocationCameraController
+import java.util.concurrent.ExecutionException
+import kotlin.math.roundToInt
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.viewshed_seek_bars.*
+
+
+class MainActivity : AppCompatActivity() {
+
+  companion object {
+    private val TAG: String = MainActivity::class.java.simpleName
+  }
+
+  private val initHeading = 0
+  private val initPitch = 60
+  private val initHorizontalAngle = 75
+  private val initVerticalAngle = 90
+  private val initMinDistance = 0
+  private val initMaxDistance = 1500
+
+  private var minDistance: Int = 0
+  private var maxDistance: Int = 0
+
+  private lateinit var viewShed: LocationViewshed
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a surface for elevation data
+    val localElevationImageService = getString(R.string.elevation_service)
+    val surface = Surface().apply {
+      elevationSources.add(ArcGISTiledElevationSource(localElevationImageService))
+    }
+
+    // create a layer of buildings
+    val buildingsLayer = ArcGISSceneLayer(getString(R.string.buildings_layer))
+
+    // create a scene and add imagery basemap, elevation surface, and buildings layer to it
+    val scene = ArcGISScene().apply {
+      basemap = Basemap.createImagery()
+      baseSurface = surface
+      operationalLayers.add(buildingsLayer)
+    }
+
+    val initLocation = Point(-4.50, 48.4, 1000.0)
+    Viewshed.setFrustumOutlineColor(Color.BLUE)
+
+    // create viewshed from the initial location
+    viewShed = LocationViewshed(
+      initLocation,
+      initHeading.toDouble(),
+      initPitch.toDouble(),
+      initHorizontalAngle.toDouble(),
+      initVerticalAngle.toDouble(),
+      initMinDistance.toDouble(),
+      initMaxDistance.toDouble()
+    ).apply {
+      setFrustumOutlineVisible(true)
+    }
+
+    // add a camera and set it to orbit the location point of the frustum
+    val camera = Camera(initLocation, 20000000.0, 0.0, 55.0, 0.0)
+    val orbitCamera = OrbitLocationCameraController(initLocation, 5000.0)
+    sceneView.apply {
+      this.scene = scene
+      this.cameraController = orbitCamera
```

Comment:
Possibly easier to just inline `Camera(initLocation, 20000000.0, 0.0, 55.0, 0.0)`

Reply:
did a bit more cleanup here by renaming scene to something more meaningful (buildingsScene) which also allowed me to drop the this. from the calls in the scope function